### PR TITLE
Add aggregate functions to calculate dependencies between categorial values

### PR DESCRIPTION
**Pull request by antikvist**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added several new aggregate functions that calculates some measure of association between two columns with categorical features as their values.

Detailed description / Documentation draft:

Added Theil's U, Contigiency Coefficient and two different types of Cramer's V
All functions show how strongly the values of one column depend on the other.
For example, Cramer's V of the same solumn equals to 1.

This closes #24537.

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Please provide any paper where you took there formulas for reference.

File: src/AggregateFunctions/registerAggregateFunctions.cpp [hunk 1]
```diff
@@ -48,6 +48,10 @@ void registerAggregateFunctionRankCorrelation(AggregateFunctionFactory &);
 void registerAggregateFunctionMannWhitney(AggregateFunctionFactory &);
 void registerAggregateFunctionWelchTTest(AggregateFunctionFactory &);
 void registerAggregateFunctionStudentTTest(AggregateFunctionFactory &);
+void registerAggregateFunctionCramersV(AggregateFunctionFactory &);
+void registerAggregateFunctionTheilsU(AggregateFunctionFactory &);
+void registerAggregateFunctionContingencyCoefficient(AggregateFunctionFactory &);
+void registerAggregateFunctionCramersVBiasCorrection(AggregateFunctionFactory &);
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/registerAggregateFunctions.cpp:54]: BiasCorrection -> BiasCorrected

**Comment by alexey-milovidov** [comment]
Link to Wikipedia is also Ok.

File: src/AggregateFunctions/AggregateFunctionContingencyCoefficient.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/AggregateFunctionCramersV.h>
+#include <AggregateFunctions/FactoryHelpers.h>
+#include <AggregateFunctions/Helpers.h>
+#include "registerAggregateFunctions.h"
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/AggregateFunctionContingencyCoefficient.cpp:5]: No need for this header.

File: src/AggregateFunctions/AggregateFunctionCramersV.h [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                pairs[hash1] += count;
+            }
+        }
+
+        void serialize(WriteBuffer &buf) const
+        {
+            writeBinary(cur_size, buf);
+            n_i.write(buf);
+            n_j.write(buf);
+            pairs.write(buf);
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/AggregateFunctionCramersV.h:71]: We can lower the amount of data passed over network if we only serialize pairs and reconstruct n_i and n_j during deserialization.

**Comment by alexey-milovidov** [comment]
Please address review comments from here: https://github.com/ClickHouse/ClickHouse/pull/24537

**Comment by alexey-milovidov** [comment]
Continued in #33366.

**Pull request closed without merging**